libs/model/src/subscription/GetSubscriptionPreferences.query.ts

```diff
@@ -0,0 +1,21 @@
+import { schemas, type Query } from '@hicommonwealth/core';
+import { models } from '../database';
+
+export const GetSubscriptionPreferences: Query<
+  typeof schemas.queries.GetSubscriptionPreferences
+> = () => ({
+  ...schemas.queries.GetSubscriptionPreferences,
+  auth: [],
+  secure: true,
+  body: async ({ actor }) => {
+    const subPreferences = await models.SubscriptionPreference.findOne({
+      where: { user_id: actor.user.id },
+    });
+
+    // We discussed throwing errors from here previously -> if not here then where can this be thrown?
```

Comment:
the proposed convention is to return an empty object or empty array or undefined instead of errors, depending on the expected results... the client app should decide what to do with empty results

Reply:
Yes but again this ignores the fact that the data should exist when queried... I believe what we discussed was reporting the error to Rollbar but returning an empty result to the client? Otherwise, we silently fail and never know there is a failure since we don't report client errors to Rollbar.

Reply:
should we consider the option where users can try to query data that might not exist yet?

Reply:
In the case when the data _should_ exist yes. If a user exists then their preferences must exist as well. Any deviation from that fact is an error/corruption.

Reply:
ok, so maybe we should create a utility guard to log the error and return empty... similar to mustExists but just to log?

Reply:
I added a `shouldExist` guard which logs an InvalidState error (but doesn't throw) if the preferences don't exist